File: Python/All_Paths_From_Source_to_Target.py
class Solution(object):
    def allPathsSourceTarget(self, graph):
        """
        :type graph: List[List[int]]
        :rtype: List[List[int]]
        """
        paths = self.pathsToTarget(graph, 0)
        for path in paths:
            path.reverse()
        return paths
        
    def pathsToTarget(self, graph, src):
        res = []
        if src == len(graph) - 1:
            res.append([src])
        elif len(graph[src]) > 0:
            for adj in graph[src]:
                paths = self.pathsToTarget(graph, adj)
                for path in paths:
                    path.append(src)
                    res.append(path)
        return res

File: Go/All_Paths_From_Source_to_Target.go
func allPathsSourceTarget(graph [][]int) [][]int {
    paths := allPathsToTarget(graph, 0)
    var tmp int
    for _, path := range(paths) {
        for i := 0; i < len(path)/2; i ++ {
            tmp = path[i]
            path[i] = path[len(path)-i-1]
            path[len(path)-i-1] = tmp
        }
    }
    return paths
}

func allPathsToTarget(graph [][]int, src int) [][]int{
    var res [][]int
    if src == len(graph) - 1 {
        res = append(res, []int{src})
    } else {
        for _, adj := range(graph[src]) {
            paths := allPathsToTarget(graph, adj)
            for _, path := range(paths) {
                res = append(res, append(path, src))
            }
        }
    }
    
    return res
}
